pkg/internal/cyberark/identity/authenticated_http_client.go

```diff
@@ -0,0 +1,19 @@
+package identity
+
+import (
+	"fmt"
+	"net/http"
+)
+
+func (c *Client) AuthenticateRequest(req *http.Request) error {
+	c.tokenCachedMutex.Lock()
+	defer c.tokenCachedMutex.Unlock()
+
+	if len(c.tokenCached) == 0 {
+		return fmt.Errorf("no token cached")
+	}
+
+	req.Header.Set("Authorization", fmt.Sprintf("Bearer %s", string(c.tokenCached)))
+
```

Comment:
suggestion (non-blocking): I think this would be fine in the existing identity.go file, but that's a matter of taste and it's not important.